pkg/credentialprovider/plugin/examples/file/main.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// file is a simple example of a docker auth plugin that fetches
+// the registry username/password by reading files on disk.
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"k8s.io/klog/v2"
+	plugin "k8s.io/kubernetes/pkg/credentialprovider/plugin/framework"
```

Comment:
We should probably build the sample outside of K/K and put it in something like staging/cloud-provider so we can ensure that the dependencies are reasonable for an external implementation.